src/test/shell/bazel/cc_integration_test.sh

```diff
@@ -1708,4 +1708,51 @@ EOF
   expect_log "in pkg/library.cpp: ''"
 }
 
+function test_compiler_flag_gcc() {
+  # The default macOS toolchain always uses XCode's clang.
+  [ "$PLATFORM" != "darwin" ] || return 0
+  type -P gcc || return 0
+
+  cat > BUILD.bazel <<'EOF'
+config_setting(
+    name = "gcc_compiler",
+    flag_values = {"@bazel_tools//tools/cpp:compiler": "gcc"},
+)
+
+cc_binary(
+  name = "main",
+  srcs = select({":gcc_compiler": ["main.cc"]}),
+)
+EOF
+  cat > main.cc <<'EOF'
+int main() {}
+EOF
+
+  bazel build //:main --repo_env=CC=gcc || fail "Expected compiler flag to have value 'gcc'"
+}
+
+function test_compiler_flag_clang() {
+  # TODO: The default toolchain always uses XCode's clang, but sets the compiler name to the generic
```

Comment:
If you force using the C++ only toolchain on macOS this would likely pass, although maybe it still shouldn't

Reply:
I briefly went over the control flow and it seemed as if the fallback to the Unix toolchain would only be picked if Xcode couldn't be find. I have a follow-up PR where I will dig deeper into the macOS case.